### PR TITLE
policies graduation to v1

### PR DESCRIPTION
**Pull request by aryan16**
**Please provide a description of this PR:**
This PR provides the support for policies graduation to V1 and updates all the security tests to use v1 version of authorization and requestAuth policies.

### CODE REVIEW

File: pilot/pkg/security/authz/builder/testdata/http/allow-empty-rule-in.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-apiVersion: security.istio.io/v1beta1
+apiVersion: security.istio.io/v1
```
Comment (howardjohn) [inline pilot/pkg/security/authz/builder/testdata/http/allow-empty-rule-in.yaml:1]: These don't need to change; current prior art from networking is to use the oldest version everywhere (its strictly better - same schema but more compatible, and requires no noise from changing a bunch of files)

File: tests/integration/pilot/validation_test.go [hunk 2]
```diff
@@ -173,6 +173,8 @@ func TestEnsureNoMissingCRDs(t *testing.T) {
 				"networking.istio.io/v1beta1/VirtualService",
 				"networking.istio.io/v1beta1/WorkloadEntry",
 				"networking.istio.io/v1beta1/Sidecar",
+				"security.istio.io/v1/AuthorizationPolicy",
```
Comment (howardjohn) [inline tests/integration/pilot/validation_test.go:176]: nit: use the `aliases` part of `r` to do this automatically; the networking stuff can be cleaned up as well (we didn't have `aliases` at the time

File: tests/integration/pilot/validation_test.go [hunk 3]
```diff
@@ -166,21 +166,22 @@ func TestEnsureNoMissingCRDs(t *testing.T) {
...
+				for _, alias := range r.Resource().GroupVersionAliasKinds() {
+					s = strings.Join([]string{alias.Group, alias.Version, alias.Kind}, "/")
+					recognized.Insert(s)
+				}
 			}
 			// These CRDs are validated outside of Istio
 			for _, gvk := range []string{
+				"networking.istio.io/v1beta1/ServiceEntry",
+				"networking.istio.io/v1beta1/WorkloadGroup",
+				"networking.istio.io/v1beta1/EnvoyFilter",
```
Comment (howardjohn) [inline tests/integration/pilot/validation_test.go:178]: why are we removing these? they shouldn't be in `recognized` anyways since they don't exist...?

**Review by howardjohn — APPROVED** [review]

File: tests/integration/pilot/validation_test.go (hunk 3 above)
Comment (howardjohn) [reply tests/integration/pilot/validation_test.go:178]: oh we should add tests then. But Envoyfilter v1beta1  is definitely wrong, we should make sure the metadata.yaml is correct

Comment (aryan16) [reply tests/integration/pilot/validation_test.go:178]: I see `v1beta1` alias for EnvoyFilter as well

Comment (howardjohn) [reply tests/integration/pilot/validation_test.go:178]: I nkow the alias is there, but the metadatal.yaml is wrong; EnvoyFilter v1beta1 doesn't exist.

**Review by zirain — APPROVED** [review]

**Pull request merged**